### PR TITLE
feat: support company-scoped plugin background workflows

### PR DESCRIPTION
**Pull request by tiangao88**
## Thinking Path

> - Plugin bridge authorization and plugin-owned action parameters are separate concerns.
> - Company scope should travel in host-issued context rather than overwrite a plugin's payload.
> - Background plugin work has no user request to supply that context, so only narrowly derived company and instance scopes should be allowed.
> - Approval callbacks also need trusted external attribution without accepting arbitrary caller-supplied identities.

## Linked Issues or Issue Description

- Refs #6057
- Related: #6148

## What Changed

- Preserve plugin-owned action parameters while passing authorized company scope through host-issued actor context.
- Allow background plugin RPC calls to derive scope only from supported sources:
  - explicit company context for company-scoped config
  - company-scoped state operations
  - instance-scoped state operations
  - instance-level `companies.list`
- Reject missing, stale, forged, and cross-company invocation scopes before host services run.
- Allow board API key integrations to record bounded `telegram:<actor>` approval attribution while session callers continue to use the authenticated board identity.
- Add focused coverage for bridge parameter preservation, background scope derivation, cross-company rejection, approval attribution, and payload validation.

## Verification

- `pnpm --filter @paperclipai/shared typecheck`
- `pnpm --filter @paperclipai/plugin-sdk typecheck`
- `pnpm --filter @paperclipai/server typecheck`
- `pnpm --filter @paperclipai/shared build`
- `pnpm --filter @paperclipai/plugin-sdk build`
- `pnpm --filter @paperclipai/server build`
- Focused non-network suites: 24/24 passed
  - `packages/plugins/sdk/tests/worker-rpc-host.test.ts`
  - `packages/shared/src/validators/approval.test.ts`
  - `server/src/__tests__/plugin-worker-manager.test.ts`

Local route-suite limitation:

- `plugin-routes-authz.test.ts` and `approval-routes-idempotency.test.ts` cannot execute on this host because Supertest binds IPv6 loopback and the local sandbox rejects `::1` with `EADDRNOTAVAIL` before assertions.
- GitHub CI runs both serialized route suites in its normal network environment.

## Risks

- Background access remains intentionally narrow; arbitrary host methods do not gain unscoped access.
- Explicit company scope is rejected when it conflicts with an active invocation.
- External approval attribution is accepted only from board API key actors and only for non-empty, single-line `telegram:` identities.

## Checklist

- [x] I have included a thinking path that traces from project context to this change
- [x] I have specified the model used (with version and capability details)
- [x] I have checked ROADMAP.md and confirmed this PR does not duplicate planned core work
- [x] I have searched GitHub for duplicate or related PRs and linked them above
- [x] I have linked the relevant issue
- [x] I have run applicable local tests, typechecks, and builds
- [x] I have added or updated tests where applicable
- [x] This change has no UI surface requiring screenshots
- [x] No documentation change is required for this internal runtime behavior
- [x] I have considered and documented risks
- [ ] All Paperclip CI gates are green
- [ ] Greptile is 5/5 with no open P2s, recommendations, or follow-ups
- [x] I will address all Greptile and reviewer comments before requesting merge

## Model Used

Codex GPT-5.6, with repository inspection, conflict resolution, local test execution, typechecking, builds, and GitHub CLI assistance.

### CODE REVIEW

File: server/src/services/plugin-secrets-handler.ts [hunk 1]
```diff
@@ -199,7 +207,9 @@ function createRateLimiter(maxAttempts: number, windowMs: number) {
 export function createPluginSecretsHandler(
   options: PluginSecretsHandlerOptions,
 ): PluginSecretsService {
-  const { pluginId } = options;
+  const { pluginId, manifest } = options;
+  const registry = pluginRegistryService(options.db);
+  const secrets = secretService(options.db);
 
   // Rate limit: max 30 resolution attempts per plugin per minute
   const rateLimiter = createRateLimiter(30, 60_000);
```
[the remark this answers is not in the document]
Comment (tiangao88) [reply server/src/services/plugin-secrets-handler.ts]: Addressed in `150bfe99`.

The resolver now keys the rate limiter by `pluginId + companyId` instead of `pluginId` alone:

```ts
const rateLimitKey = `${pluginId}:${companyId || "__no_company__"}`;
```

I also added coverage for this case in `plugin-secrets-handler-runtime.test.ts`: one company can exhaust its own bucket without blocking another company using the same plugin.

File: server/src/routes/plugins.ts [hunk 2]
```diff
@@ -1075,15 +1097,15 @@ export function pluginRoutes(
...
+    const companyId = assertPluginBridgeScope(req, body.companyId);
 
     try {
       const result = await bridgeDeps.workerManager.call(
         plugin.id,
         "getData",
         {
           key: body.key,
-          params: body.params ?? {},
+          params: companyId ? { ...(body.params ?? {}), companyId } : body.params ?? {},
```
[the remark this answers is not in the document]
Comment (tiangao88) [reply server/src/routes/plugins.ts]: Addressed in `307dbb0e`.

The runtime now carries `companyId` out-of-band through the worker context and strips it before invoking plugin handlers, so plugin data/action handlers do not receive an unexpected `companyId` field in their params.

This keeps the company context available for `ctx.config.get()` / `ctx.secrets.resolve()` while preserving compatibility with plugins that validate params strictly.

File: server/src/services/plugin-secrets-handler.ts [hunk 3]
```diff
@@ -230,9 +240,32 @@ export function createPluginSecretsHandler(
...
+        configRow?.configJson ?? {},
+        manifest?.instanceConfigSchema,
+      );
+      const paths = [...(refsBySecret.get(trimmedRef) ?? [])];
+      if (paths.length === 0) {
+        throw new Error("Secret is not referenced by this company's plugin config");
+      }
+      if (paths.length > 1) {
+        throw new Error("Secret reference is ambiguous in this company's plugin config");
+      }
```
[the remark this answers is not in the document]
Comment (tiangao88) [reply server/src/services/plugin-secrets-handler.ts]: Addressed in `150bfe99`.

The ambiguity error now includes the duplicate schema-declared config paths, e.g.:

```text
Secret reference is ambiguous in this company's plugin config at: apiKeyRef, webhookKeyRef
```

I also added a regression test covering duplicate `secret-ref` paths so operators get actionable remediation guidance instead of a generic ambiguity error.

**Comment by tiangao88** [comment]
Thanks for flagging. I restricted Telegram actor overrides to integration-authenticated calls only (`board_key`) and added coverage for ordinary authenticated callers falling back to their own user id. Superagent is now green on the latest #6173 head.

**Comment by tiangao88** [comment]
CI status after rebasing and cleaning this PR against current `master`:

- The PR-specific plugin route/worker failures are resolved on `20be17e4f`.
- Serialized server suite 3 now passes.
- Greptile, Superagent, build, typecheck/release registry, Canary dry run, and serialized suite 4 pass.
- Serialized server suite 2 remains blocked by the current `master` baseline: `authz-existence-oracle-guard.test.ts` reports five untouched sites in `server/src/routes/tool-access.ts`.

The same suite 2 failure occurs in the Release workflow on current base commit `7f2ed0ad9` (run 29374424448). This PR does not modify `tool-access.ts` or the existence-oracle guard test.
